src/operator/mxnet_op.h

```diff
@@ -448,6 +449,47 @@ struct Kernel<OP, cpu> {
 #endif
   }
 
+  /*!
+   * \brief Launch a generic CPU kernel with native random generator.
+   * \tparam rnd CPU random generator
+   * \tparam Args Varargs type to eventually pass to the OP::Map() functoion
+   * \param N Number of iterations
+   * \param args Varargs to eventually pass to the OP::Map() functoion
+   */
+  template<typename GType, typename ...Args>
+  inline static void LaunchRNG(mshadow::Stream<cpu> *,
+                               common::random::RandGenerator<cpu, GType> *rnd,
+                               const int N, Args... args) {
+    using namespace mxnet::common::random;
+#ifdef _OPENMP
```

Comment:
I don't think you need that ifdef. Even if we compile w/out openmp-support, the functions GetRecommendedOMPTHreadCount() and omp_get_thread_num() should be replaced by appropriate stubs. So your code does not need to do any specialties (see dmlc/omp.h).